File: samples/simple3Dscan.py
#a simple 3d scan from an image

import cv2
import open3d as o3d

# capture depth image using OpenCV
depth_image = cv2.imread('depth_image.png', cv2.IMREAD_UNCHANGED)

# convert depth image to point cloud using Open3D
depth = o3d.geometry.Image(depth_image)
intrinsic = o3d.camera.PinholeCameraIntrinsic(
    depth_image.shape[1], depth_image.shape[0], fx=525, fy=525, cx=319.5, cy=239.5)
pcd = o3d.geometry.PointCloud.create_from_depth_image(depth, intrinsic)

# create mesh from point cloud using Open3D
mesh = o3d.geometry.TriangleMesh.create_from_point_cloud_poisson(pcd)

# export mesh as STL file using Open3D
o3d.io.write_triangle_mesh("mesh.stl", mesh)

File: data/samples/simple3dscan.cpp
#include <opencv2/opencv.hpp>
#include <open3d/Open3D.h>

int main() {
    // capture depth image using OpenCV
    cv::Mat depth_image = cv::imread("depth_image.png", cv::IMREAD_UNCHANGED);

    // convert depth image to point cloud using Open3D
    auto depth = open3d::geometry::Image::CreateDepthImageFromNumpy(depth_image);
    open3d::camera::PinholeCameraIntrinsic intrinsic(depth_image.cols, depth_image.rows, 525, 525, 319.5, 239.5);
    auto pcd = open3d::geometry::PointCloud::CreateFromDepthImage(depth, intrinsic);

    // create mesh from point cloud using Open3D
    auto mesh = open3d::geometry::TriangleMesh::CreateFromPointCloudPoisson(*pcd);

    // export mesh as STL file using Open3D
    open3d::io::WriteTriangleMesh("mesh.stl", *mesh);
}
